astronomer/providers/snowflake/sensors/snowflake.py

```diff
@@ -74,30 +75,61 @@ def __init__(
     def execute(self, context: Context) -> None:
         """Check for query result in Snowflake by deferring using the trigger"""
         if not poke(self, context):
-            self.defer(
-                timeout=timedelta(seconds=self.timeout),
-                trigger=SnowflakeSensorTrigger(
-                    sql=self.sql,
-                    poke_interval=self.poke_interval,
-                    parameters=self.parameters,
-                    success=self.success,
-                    failure=self.failure,
-                    fail_on_empty=self.fail_on_empty,
-                    dag_id=context["dag"].dag_id,
-                    task_id=context["task"].task_id,
-                    run_id=context["dag_run"].run_id,
-                    snowflake_conn_id=self.snowflake_conn_id,
-                ),
-                method_name=self.execute_complete.__name__,
-            )
+            self._defer(context)
```

Comment:
why is private method needed here?

Reply:
I prefer private method because it easy to remove in future without deprecation